### ISSUE TITLE
package Errors

### ISSUE THREAD

**Issue by denisok** [openSUSE/openSUSEway#92]
while building a package there are some errors reported:

```
[    6s] RPMLINT report:
[    6s] ===============
[    7s] ============================ rpmlint session starts ============================
[    7s] rpmlint: 2.4.0
[    7s] configuration:
[    7s]     /opt/testing/lib64/python3.10/rpmlint/configdefaults.toml
[    7s]     /opt/testing/share/rpmlint/cron-whitelist.toml
[    7s]     /opt/testing/share/rpmlint/dbus-services.toml
[    7s]     /opt/testing/share/rpmlint/device-files-whitelist.toml
[    7s]     /opt/testing/share/rpmlint/licenses.toml
[    7s]     /opt/testing/share/rpmlint/opensuse.toml
[    7s]     /opt/testing/share/rpmlint/pam-modules.toml
[    7s]     /opt/testing/share/rpmlint/permissions-whitelist.toml
[    7s]     /opt/testing/share/rpmlint/pie-executables.toml
[    7s]     /opt/testing/share/rpmlint/polkit-rules-whitelist.toml
[    7s]     /opt/testing/share/rpmlint/scoring.toml
[    7s]     /opt/testing/share/rpmlint/security.toml
[    7s]     /opt/testing/share/rpmlint/sudoers-whitelist.toml
[    7s]     /opt/testing/share/rpmlint/sysctl-whitelist.toml
[    7s]     /opt/testing/share/rpmlint/systemd-tmpfiles.toml
[    7s]     /opt/testing/share/rpmlint/users-groups.toml
[    7s]     /opt/testing/share/rpmlint/world-writable-whitelist.toml
[    7s]     /opt/testing/share/rpmlint/zypper-plugins.toml
[    7s] checks: 41, packages: 5
[    7s] 
[    7s] sway-branding-openSUSE.noarch: E: systemd-service-without-service_del_preun sway-session.target
[    7s] sway-branding-openSUSE.noarch: E: systemd-service-without-service_del_preun sway.service
[    7s] The package contains a systemd service but doesn't contain a %preun with a
[    7s] call to service_del_preun.
[    7s] 
[    7s] sway-branding-openSUSE.noarch: E: systemd-service-without-service_del_postun sway-session.target
[    7s] sway-branding-openSUSE.noarch: E: systemd-service-without-service_del_postun sway.service
[    7s] The package contains a systemd service but doesn't contain a %postun with a
[    7s] call to service_del_postun.
[    7s] 
[    7s] sway-branding-openSUSE.noarch: E: systemd-service-without-service_add_pre sway-session.target
[    7s] sway-branding-openSUSE.noarch: E: systemd-service-without-service_add_pre sway.service
[    7s] The package contains a systemd service but doesn't contain a %pre with a call
[    7s] to service_add_pre.
[    7s] 
[    7s] sway-branding-openSUSE.noarch: E: systemd-service-without-service_add_post sway-session.target
[    7s] sway-branding-openSUSE.noarch: E: systemd-service-without-service_add_post sway.service
[    7s] The package contains a systemd service but doesn't contain a %post with a call
[    7s] to service_add_post.
[    7s] 
[    7s] sway-branding-openSUSE.noarch: E: suse-zypp-packageand packageand(sway:branding-openSUSE)
[    7s] waybar-branding-openSUSE.noarch: E: suse-zypp-packageand packageand(waybar:branding-openSUSE)
[    7s] The 'packageand(package1:package2)' syntax is obsolete, please use boolean
[    7s] dependencies like: 'Supplements: (package1 and package2)'
[    7s] 
[    7s] sway-branding-openSUSE.noarch: E: suse-zypp-otherproviders otherproviders(sway-branding)
[    7s] waybar-branding-openSUSE.noarch: E: suse-zypp-otherproviders otherproviders(waybar-branding)
[    7s] The 'otherproviders(symbol)' syntax is obsolete, it is not needed and you can
[    7s] use the 'symbol' directly: 'Conflicts: symbol'
[    7s] 
[    7s] openSUSEway.noarch: W: non-standard-uid /etc/greetd/config.toml.way greeter
[    7s] openSUSEway.noarch: W: non-standard-uid /etc/greetd/environments greeter
[    7s] openSUSEway.noarch: W: non-standard-uid /etc/greetd/sway-config greeter
[    7s] A file in this package is owned by an unregistered user id. To register the
[    7s] user, please make a pull request to the rpmlint config file
[    7s] configs/openSUSE/users-groups.toml in the rpmlint repository.
[    7s] 
[    7s] openSUSEway.noarch: W: non-standard-gid /etc/greetd/config.toml.way greeter
[    7s] openSUSEway.noarch: W: non-standard-gid /etc/greetd/environments greeter
[    7s] openSUSEway.noarch: W: non-standard-gid /etc/greetd/sway-config greeter
[    7s] A file in this package is owned by an unregistered group id. To register the
[    7s] group, please make a pull request to the rpmlint config file
[    7s] configs/openSUSE/users-groups.toml in the rpmlint repository.
[    7s] 
[    7s] openSUSEway.noarch: W: name-repeated-in-summary openSUSEway
[    7s] openSUSEway.src: W: name-repeated-in-summary openSUSEway
[    7s] The name of the package is repeated in its summary. Make the summary brief and
[    7s] to the point without including redundant information in it.
[    7s] 
[    7s] sway-branding-openSUSE.noarch: E: branding-supplements-missing (sway and branding-openSUSE)
[    7s] waybar-branding-openSUSE.noarch: E: branding-supplements-missing (waybar and branding-openSUSE)
[    7s] Branding packages should provide a supplement in the form: 'Supplements:
[    7s] (basepackage and branding-<flavor>)'
[    7s] 
[    7s] openSUSEway.noarch: E: branding-requires-specific-flavor gfxboot-branding-openSUSE
[    7s] openSUSEway.noarch: E: branding-requires-specific-flavor sway-branding-openSUSE
[    7s] openSUSEway.noarch: E: branding-requires-specific-flavor waybar-branding-openSUSE
[    7s] Packages must not require a specific branding or theme package to allow for
[    7s] different themes.
[    7s] 
[    7s] sway-branding-openSUSE.noarch: E: branding-conflicts-missing sway-branding
[    7s] waybar-branding-openSUSE.noarch: E: branding-conflicts-missing waybar-branding
[    7s] Branding packages should conflict with other flavors of the branding package
[    7s] by using: 'Conflicts: pkg-branding = brandingversion' and not directly by
[    7s] listing all the alternative brandings in it.
[    7s] 
[    7s] Check time report (>1% & >0.1s):
[    7s]     Check                            Duration (in s)   Fraction (in %)  Checked files
[    7s]     TOTAL                                        0.2             100.0             30
[    7s] 
[    7s]  5 packages and 0 specfiles checked; 19 errors, 8 warnings, 32 filtered, 19 badness; has taken 0.2 s 
```

**Comment by FilippoBonazziSUSE** [comment]
I'm looking at these. Since the spec file is tracked on Github now, I am thinking of doing this:

1. Fix the issues in the OBS copy of the spec file, thus triggering 0.15-N builds on OBS
2. Apply the same fixes here on GH, so that when we make a new release the spec file is updated.

Alternatively I could do the spec file changes directly on GH, but then I would have to create new releases on GH and resubmit to OBS (0.15.N)

**Comment by denisok** [comment]
@FilippoBonazziSUSE those issues you could also clearly see doing build locally with `osc build` so you could just fix it locally and submit new PR here. No need to add OBS into it :)

Usually my workflow:
- branch a package in OBS (if I would like to submit new version) or just use X11:Wayland
- checkout with `osc co`
- make changes
- build with `osc build`
- install new resulting packages to verify they are OK
- submit PR here or commit to OBS and submit it to the X11:Wayland

**Comment by FilippoBonazziSUSE** [comment]
This is still open as some packaging errors and warnings are still unaddressed:

```
[   14s] openSUSEway.noarch: W: non-standard-uid /etc/greetd/config.toml.way greeter
[   14s] openSUSEway.noarch: W: non-standard-uid /etc/greetd/environments greeter
[   14s] openSUSEway.noarch: W: non-standard-uid /etc/greetd/style.css greeter
[   14s] openSUSEway.noarch: W: non-standard-uid /etc/greetd/sway-config greeter
[   14s] A file in this package is owned by an unregistered user id. To register the
[   14s] user, please make a pull request to the rpmlint config file
[   14s] configs/openSUSE/users-groups.toml in the rpmlint repository.
[   14s]
[   14s] openSUSEway.noarch: W: non-standard-gid /etc/greetd/config.toml.way greeter
[   14s] openSUSEway.noarch: W: non-standard-gid /etc/greetd/environments greeter
[   14s] openSUSEway.noarch: W: non-standard-gid /etc/greetd/style.css greeter
[   14s] openSUSEway.noarch: W: non-standard-gid /etc/greetd/sway-config greeter
[   14s] A file in this package is owned by an unregistered group id. To register the
[   14s] group, please make a pull request to the rpmlint config file
[   14s] configs/openSUSE/users-groups.toml in the rpmlint repository.
```
I could look into these ones.
```
[   14s]
[   14s] openSUSEway.noarch: W: name-repeated-in-summary openSUSEway
[   14s] openSUSEway.src: W: name-repeated-in-summary openSUSEway
[   14s] The name of the package is repeated in its summary. Make the summary brief and
[   14s] to the point without including redundant information in it.
```
I think this one can safely be ignored.
```
[   14s]
[   14s] openSUSEway.noarch: E: branding-requires-specific-flavor gfxboot-branding-openSUSE
[   14s] openSUSEway.noarch: E: branding-requires-specific-flavor sway-branding-openSUSE
[   14s] openSUSEway.noarch: E: branding-requires-specific-flavor waybar-branding-openSUSE
[   14s] Packages must not require a specific branding or theme package to allow for
[   14s] different themes.
```
This one's tricky, I think I accidentally broke something before by requiring versioned branding. So I think this should probably stay as it is, but it would need some investigation.

**Comment by FilippoBonazziSUSE** [comment]
```
[   14s] openSUSEway.noarch: W: non-standard-uid /etc/greetd/config.toml.way greeter
[   14s] openSUSEway.noarch: W: non-standard-uid /etc/greetd/environments greeter
[   14s] openSUSEway.noarch: W: non-standard-uid /etc/greetd/style.css greeter
[   14s] openSUSEway.noarch: W: non-standard-uid /etc/greetd/sway-config greeter
[   14s] A file in this package is owned by an unregistered user id. To register the
[   14s] user, please make a pull request to the rpmlint config file
[   14s] configs/openSUSE/users-groups.toml in the rpmlint repository.
[   14s]
[   14s] openSUSEway.noarch: W: non-standard-gid /etc/greetd/config.toml.way greeter
[   14s] openSUSEway.noarch: W: non-standard-gid /etc/greetd/environments greeter
[   14s] openSUSEway.noarch: W: non-standard-gid /etc/greetd/style.css greeter
[   14s] openSUSEway.noarch: W: non-standard-gid /etc/greetd/sway-config greeter
[   14s] A file in this package is owned by an unregistered group id. To register the
[   14s] group, please make a pull request to the rpmlint config file
[   14s] configs/openSUSE/users-groups.toml in the rpmlint repository.
```

Registered the `greeter` uid and gid in https://github.com/rpm-software-management/rpmlint/pull/1184

**Comment by FilippoBonazziSUSE** [comment]
For the record, after #187 and #193 we should be down to zero visible errors/warnings.
I will keep checking if more appear in the future.
